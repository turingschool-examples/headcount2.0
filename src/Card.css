
.card {
  width: 150px;
  text-align: center;
  margin: 2%;
  background: white;
  opacity: .86;
  border-radius: 5px;
  padding: 10px;
  box-shadow: 3px 3px 3px #7c7c7c;
}

h1 {
  font-size: 20px;
}

.data {
  font-size: 15px;
  text-align: left;
  font-weight: bold;
}

.less-than {
  color: #ea6512;
}

.greater-than {
  color: #4f4f4f;
}
