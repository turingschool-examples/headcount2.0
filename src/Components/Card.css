.card {
  background-color: #375053;
  border-radius: 15px;
  box-shadow: 0px 0px 19px 1px rgba(0, 0, 0, .7);
  flex: 1 18em;
  list-style-type: none;
  margin: 0 2em 2em 2em;
  padding: 0 1em 1em 1em;
}

.selected {
  background-color: #5e7377;
  border: solid 4px #f7b488;
}

h2 {
  font-size: 2em;
  margin: .5em;
  text-align: center;
}

li {
  font-size: 1.2em;
  padding: 0 0 0 6em;
}

.stat {
  padding: 0 0 .5em 0;
}

.less-than {
  color: #f7b488;
}