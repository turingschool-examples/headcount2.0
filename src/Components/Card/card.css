.card-div{
  border: 1.3px solid white;
  width: 17em;
  height: 33em;
  margin: 0 auto;
  word-wrap: break-word;
  text-align: center;
  padding: 1em;
  margin-bottom: 2em;
  border-radius: 10px;
  box-shadow: 6px 6px 6px rgba(0, 0, 0, 0.5);
  background-color: #5c687c;
  flex-basis: 20%;
}

.location{
  color: white;
}

.district-data{
  color: #cacdd1;
}

.year{
  color: black;
  font-weight: bold;
}

.above50{
  color: #404142;
}

.below50{
  color: white;
}
