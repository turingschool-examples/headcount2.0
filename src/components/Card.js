import React from 'react';
import PropTypes from 'prop-types';


const Card = ({ location, data, handleCompare, compare, active }) => {
  const classname = active === true ? 'card active' : 'card'
  return (
    <div className={classname} key={location} onClick={() => handleCompare(location)}>
      <h1 className='cardTitle'>
        {location}
      </h1>
      {Object.keys(data).map((key, index) => {


        return (
          <div className='cardData' key={index}>
            <li className={
              data[key] <= 0.5 ? 'negative' : 'positive'
            }> {key}: {data[key]} </li>
          </div>
        )
      })}
    </div>
  );
}

Card.propTypes = {
  location: PropTypes.string,
  data: PropTypes.object,
  handleCompare: PropTypes.func,
}

export default Card;
