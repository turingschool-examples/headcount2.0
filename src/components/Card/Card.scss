$backgroundColor: rgba(255, 255, 255, .1);
$sublimeGreen: #AAD74F;
$sublimeRed: #D23D54;
$sublimeBlue: #78C9D0;
$sublimeYellow: #FFFF74;

article {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
  width: 250px;
  border: 1px solid #fff;
  background: $backgroundColor;
}

article h3 {
  color: $sublimeGreen;
  font-size: 1.5em;
  margin-bottom: 0px;
}

article p {
  margin: 7px;
}

ul {
  text-align: left;
  list-style: none;
  padding: 0;

  li {
    margin: 3px 3px;
  }

  .over {
    color: $sublimeBlue;
  }

  .under {
    color: $sublimeRed;
  }

  .fa {
    margin-right: 8px;
  } 

  .fa-bolt {
    color: yellow;
  }

  .fa-tint {
    color: #849FED;
  }
}

.clicked {
  border: 2px solid $sublimeYellow;
}