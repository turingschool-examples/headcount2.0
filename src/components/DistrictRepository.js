import React, { Component } from "react";
import Controls from "./Controls";
import { DistrictContainer } from "./DistrictContainer";
import KinderData from "../../data/kindergartners_in_full_day_program";
import "../styles/DistrictRepository.css";

export default class DistrictRepository extends Component {
  constructor(rawData) {
    super();
    // this.state.data = this.getDistrictData();
    this.state = {
      cards: [...this.findAllMatches()],
      data: this.getDistrictData()
    };
  }

  findAllMatches(userInput) {
    if (userInput) {
      userInput = userInput.toUpperCase();
    }

    const dataSet = this.getDistrictData();
    const districtKeys = Object.keys(dataSet).map(key => {
      return key.toUpperCase();
    });

    if (!userInput) {
      return districtKeys;
    }

    const filteredMatches = districtKeys.filter(district =>
      district.includes(userInput)
    );

    return filteredMatches;
  }

  findByName(userInput) {
    if (userInput) {
      userInput = userInput.toUpperCase();
    }
    const dataSet = this.state.data;
    const districtKeys = Object.keys(dataSet).map(key => {
      return key.toUpperCase();
    });

    let districtName = dataSet[userInput];
    let answer;

    if (districtKeys.includes(userInput)) {
      return (answer = {
        location: userInput,
        data: this.getData(districtName)
      });
    }
  }

  getData(districtInfo) {
    districtInfo = districtInfo.reduce((acc, item) => {
      if (item.Data === "N/A") {
        item.Data = 0;
      }
      acc[item.TimeFrame] = parseFloat(item.Data.toFixed(3));
      return acc;
    }, {});
    return districtInfo;
  }

  getDistrictData() {
    var rawData = KinderData;

    const cleanData = rawData.reduce((acc, obj) => {
      if (!acc[obj.Location]) {
        acc[obj.Location] = [];
      }
      if (acc[obj.Location]) {
        acc[obj.Location].push(obj);
      }
      return acc;
    }, {});
    return cleanData;
  }

  // componentDidMount()

  render() {
    return (
      <div className="district-repository-container">
        <Controls findDistrict={this.findByName.bind(this)} />
        <DistrictContainer
          foundData={this.state.cards}
          fullData={this.state.data}
        />
      </div>
    );
  }
}
