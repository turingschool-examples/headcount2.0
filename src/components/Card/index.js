import React, { Component } from 'react'
import PropTypes from 'prop-types'
import './Card.css'

class Card extends Component {
  constructor({location, stats, adjustComparisons}) {
    super()
    this.state = {
      location: location,
      stats: stats,
      isSelected: false
    }
  }

  makeCards = stats => {
    const isSelected = this.state.isSelected

    return Object.keys(stats).map((year, index) => {
      return (
        <li key={index}
            className={`Card-entry ${this.checkLow(stats, year)}`}
        >
          {year}: {stats[year]}
        </li>)
      }
    )
  }

  checkLow = (stats, year) => {
    if (stats[year] <= 0.5) return 'low'
  }

  checkSelected = isSelected => {
    if (isSelected) {
      return 'selected'
    } else {
      return ''
    }
  }

  toggleSelect = event => {
    const card = event.target.parentElement.find('.Card')

    debugger

    this.setState({ isSelected: !this.state.isSelected })
    this.props.adjustComparisons(card)
  }

  render() {
    const location = this.state.location
    const stats = this.state.stats
    const isSelected = this.state.isSelected

    return(
      <article
        className={`Card ${this.checkSelected(isSelected)}`}
        onClick={this.toggleSelect}
      >
        <h2 className="Card-title">{location}</h2>
        <ul className="Card-list">
          { stats ? this.makeCards(stats) : null}
        </ul>
      </article>
    )
  }
}

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object
}

export default Card