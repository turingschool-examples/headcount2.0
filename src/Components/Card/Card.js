import React from 'react';
import './card.css'
import PropTypes from 'prop-types';

const Card = ({location, stats}) => {

  const yearsArray = Object.keys(stats);

  const districtData = yearsArray.map((year, index) => { return <p className={stats[year] > 0.5 ? 'above50' : 'below50' }
  key={index}>
  <span className="year">{year}: </span>
  {stats[year]}</p>})

  return(
    <div className="card-div">
    <section className="info-section">
    <h2 className="location">{location}</h2>
    <h3 className="district-data">{districtData}</h3>
    </section>
    </div>
  )
}

Card.propTypes = {
  stats: PropTypes.object
}

export default Card;
