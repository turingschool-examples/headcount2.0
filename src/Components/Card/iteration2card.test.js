import React from 'react';
import { mount, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Card from './Card'
import DistrictRepository from '../../helper.js';
import kinderData from '../../data/kindergartners_in_full_day_program.js';

describe('Card', () =>  {

  const mockData = {
      location: 'Turing',
      stats: {
        '2018' : 1
      }
    }

  const district = new DistrictRepository(kinderData);

  it('should match the snapshot', () => {

    const wrapper = shallow(<Card {...mockData}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('data should present as white if below 0.5, and grey if above 0.5', () => {

    const wrapper = shallow(<Card {...mockData}/>);

    if(mockData.stats < 0.5){
      expect(wrapper.find('p').hasClass('below50')).toEqual(true);
    }else{
      expect(wrapper.find('p').hasClass('above50')).toEqual(true);
    }

  });

});
