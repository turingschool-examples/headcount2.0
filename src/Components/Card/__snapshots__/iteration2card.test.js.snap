// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    location="Turing"
    stats={
      Object {
        "2018": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="info-section"
      >
        <h2
          className="location"
        >
          Turing
        </h2>
        <h3
          className="district-data"
        >
          <p
            className="above50"
          >
            <span
              className="year"
            >
              2018
              : 
            </span>
            1
          </p>
        </h3>
      </section>,
      "className": "card-div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="location"
          >
            Turing
          </h2>,
          <h3
            className="district-data"
          >
            <p
              className="above50"
            >
              <span
                className="year"
              >
                2018
                : 
              </span>
              1
            </p>
          </h3>,
        ],
        "className": "info-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Turing",
            "className": "location",
          },
          "ref": null,
          "rendered": "Turing",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="above50"
              >
                <span
                  className="year"
                >
                  2018
                  : 
                </span>
                1
              </p>,
            ],
            "className": "district-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="year"
                  >
                    2018
                    : 
                  </span>,
                  1,
                ],
                "className": "above50",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "2018",
                      ": ",
                    ],
                    "className": "year",
                  },
                  "ref": null,
                  "rendered": Array [
                    "2018",
                    ": ",
                  ],
                  "type": "span",
                },
                1,
              ],
              "type": "p",
            },
          ],
          "type": "h3",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="info-section"
        >
          <h2
            className="location"
          >
            Turing
          </h2>
          <h3
            className="district-data"
          >
            <p
              className="above50"
            >
              <span
                className="year"
              >
                2018
                : 
              </span>
              1
            </p>
          </h3>
        </section>,
        "className": "card-div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="location"
            >
              Turing
            </h2>,
            <h3
              className="district-data"
            >
              <p
                className="above50"
              >
                <span
                  className="year"
                >
                  2018
                  : 
                </span>
                1
              </p>
            </h3>,
          ],
          "className": "info-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Turing",
              "className": "location",
            },
            "ref": null,
            "rendered": "Turing",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="above50"
                >
                  <span
                    className="year"
                  >
                    2018
                    : 
                  </span>
                  1
                </p>,
              ],
              "className": "district-data",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      className="year"
                    >
                      2018
                      : 
                    </span>,
                    1,
                  ],
                  "className": "above50",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "2018",
                        ": ",
                      ],
                      "className": "year",
                    },
                    "ref": null,
                    "rendered": Array [
                      "2018",
                      ": ",
                    ],
                    "type": "span",
                  },
                  1,
                ],
                "type": "p",
              },
            ],
            "type": "h3",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
